routes.js

```diff
@@ -30,8 +30,11 @@ var auth = function (req, res, next) {
 var combineMessageInfoAndMessage = function (messages, matchMessages)
 {
   let matches = [];
-  for (i = 0; i < matchMessages.length; i++)
-  {
+  if(typeof matchMessages === 'undefined' || matchMessages === null) {
```

Comment:
I would just do `if (!matchMessages)` This will catch anything falsy, including null, undefined, empty array, etc.

Reply:
Has updated it to if (!matchMessages)